tools/tests/dropdown/dropdown-state.test.js

```diff
@@ -0,0 +1,104 @@
+        dropdown.disabled = true;
+        assert(dropdown.classList.contains('guic-dropdown-disabled') === true, 'Dropdown element is not disabled');
+        assert(dropdown.disabled === true, 'Dropdown element is not disabled');
+
+        // remove disabled
+        dropdown.disabled = false;
+        assert(dropdown.classList.contains('guic-dropdown-disabled') === false, 'Dropdown element is disabled');
+        assert(dropdown.disabled === false, 'Dropdown element is disabled');
+    });
+
+    it('Should test collapsable state of the dropdown', () => {
+        const dropdown = document.querySelector(dropdownSelector);
+
+        dropdown.collapsable = true;
+        assert(dropdown.querySelector('.guic-dropdown-header').style.display === 'flex', 'Element is not collapsable');
+        assert(dropdown.collapsable === true, 'Element is not collapsable');
+
+        dropdown.collapsable = false;
+        assert(dropdown.querySelector('.guic-dropdown-header').style.display === 'none', 'Element is collapsable');
+        assert(dropdown.collapsable === false, 'Element is collapsable');
+    });
+
+    it('Should test multiple state of the dropdown', async () => {
+        const dropdown = document.querySelector(dropdownSelector);
+        const options = dropdown.querySelectorAll('dropdown-option');
+
+        dropdown.multiple = true;
+
+        dropdown.onClickOption(mockEventObject(options[1], true, true));
+        dropdown.onClickOption(mockEventObject(options[2], true, true));
+        dropdown.onClickOption(mockEventObject(options[3], true, true));
+        // length should be 4 because of the pre-selected option
+        assert(dropdown.selectedList.length === 4, 'The dropdown did not select multiple options');
+        assert(dropdown.multiple === true, 'The dropdown has no multiple select');
+
+        dropdown.multiple = false;
+        assert(dropdown.selectedList.length === 1, 'The dropdown did not select single option');
+        assert(dropdown.multiple === false, 'The dropdown has multiple select');
+    });
+
+    it('Should test value state of the dropdown', () => {
+        const dropdown = document.querySelector(dropdownSelector);
+        const options = dropdown.querySelectorAll('dropdown-option');
+
+        dropdown.value = 'Giraffe';
+        assert(options[2].className === 'guic-dropdown-option-active', `Option value was not ${options[2].value}`);
+        assert(dropdown.value === 'Giraffe', `Option value was not ${options[2].value}`);
+    });
+
+    it('Should test invalid value state of the dropdown', () => {
+        const dropdown = document.querySelector(dropdownSelector);
+        const options = dropdown.querySelectorAll('dropdown-option');
+
+        dropdown.value = 'invalid';
+        assert(options[0].className === 'guic-dropdown-option-active', `Option 'Cat' is not selected as default one.`);
+        assert(dropdown.value === 'Cat', `Option value was not Cat, it was ${dropdown.value}`);

```

Comment:
Use as a constant for the default value. If the template changes and we forget to change it here it will fail the test.

Reply:
Added constant for the default value